google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/apis/CloudApiManager.java

```diff
@@ -162,6 +363,27 @@ private static void enableApi(
     }
   }
 
+  /**
+   * Creates the unique ID of the service account.
+   *
+   * <p>Must be less than 30 characters.
+   *
+   * @param prefix the name chosen by the user for the service account
+   * @return an ID that is a combination of the prefix and a timestamp
+   */
+  private static String createServiceAccountId(String prefix) {
+    int maxLen = 30;
+    String timestamp = getTimestamp();
+    int maxPrefixLen = maxLen - timestamp.length() - 1;
```

Comment:
UX Nit: we could possibly define max prefix length as a constant based on timestamp length and limit text field a user types service account name into to that length so they won't be surprised by trimming.

Reply:
and to add to this, it appears as though there is a restriction to the pattern that the service account name can take.

I think I'll create a new issue to update this validation (including a regex on the service account field) to fix in a separate PR.